pkg/kubelet/dockershim/exec.go

```diff
@@ -149,16 +152,28 @@ func (*NativeExecHandler) ExecInContainer(client libdocker.Interface, container
 
 	// Have to start this before the call to client.StartExec because client.StartExec is a blocking
 	// call :-( Otherwise, resize events don't get processed and the terminal never resizes.
-	kubecontainer.HandleResizing(resize, func(size remotecommand.TerminalSize) {
-		client.ResizeExecTTY(execObj.ID, int(size.Height), int(size.Width))
-	})
+	execStarted := make(chan struct{})
+	go func() {
```

Comment:
Could you add a comment here to outline, why this is async: 

"Delay attempting to send a terminal resize request to docker until after
the exec has started; otherwise, the initial resize request will fail."

Just reading the code I would lose context if I came back to this 6 months from now.